.github/workflows/cd-infrastructure.yml

```diff
@@ -9,13 +9,14 @@ on:
     branches:
       - main
     paths: ['.github/workflows/**','infrastructure/**']
```

Comment:
@decodingahmed Great idea would make our lives so much easier, but one of those annoying obstacles where I don't think its supported :(, name-prefixes i.e., terraform-1.....yaml, terraform-2.....yaml could be a work around for now, feel free to research more though in case I made a mistake in my findings: 

https://stackoverflow.com/questions/62288443/where-should-you-put-the-git-hub-workflow-directory-for-actions-in-a-full-stack
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#about-yaml-syntax-for-workflows